include/LocklessRingBuffer.h

```diff
@@ -0,0 +1,133 @@
+/*
+ * LocklessRingBuffer.h - LMMS wrapper for a lockless ringbuffer library
+ *
+ * Copyright (c) 2019 Martin Pavelek <he29/dot/HS/at/gmail/dot/com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef LOCKLESSRINGBUFFER_H
+#define LOCKLESSRINGBUFFER_H
+
+#include "lmms_export.h"
+#include "../src/3rdparty/ringbuffer/include/ringbuffer/ringbuffer.h"
+#include "lmms_basics.h"
+#include <QMutex>
+#include <QWaitCondition>
+
+
+//! A convenience layer for a realtime-safe and thread-safe multi-reader ring buffer library.
+template <class T>
+class LocklessRingBuffer
+{
+	template<class _T>
+	friend class LocklessRingBufferReader;
+public:
+	LocklessRingBuffer(std::size_t sz) : m_buffer(sz) {};
```

Comment:
As this is always used in RT context, I'd initially call `m_buffer.touch()` to write on all the memory. This should avoid memory loading while the RT thread is working.